lib/Dancer2/Core/App.pm

```diff
@@ -889,7 +889,11 @@ sub send_file {
                 [ $res->status, $res->headers_to_array, $fh ]
             );
         };
+
+        Scalar::Util::weaken( my $weak_self = $self );
+
         $response = Dancer2::Core::Response::Delayed->new(
+            error_cb => sub { $weak_self->logger_engine->log(@_) },
```

Comment:
The call to `logger_engine->log(@_)` is missing an argument. It needs both a *log level* and the log message.